pkg/cluster/correct_cert_issuer.go

```diff
@@ -78,7 +78,7 @@ func (m *manager) ensureCertificateIssuer(ctx context.Context, certificateName,
 			return err
 		}
 
-		_, err = clusterKeyvault.CreateCertificate(ctx, issuerName, azcertificates.SignedCertificateParameters(certificateName, dnsName, azcertificates.EkuServerAuth), nil)
+		_, err = clusterKeyvault.CreateCertificate(ctx, certificateName, azcertificates.SignedCertificateParameters(issuerName, dnsName, azcertificates.EkuServerAuth), nil)
```

Comment:
I think we need to move this call outside the `if *bundle.Policy.IssuerParameters.Name != issuerName` conditional, and wrap it in its own conditional that validates the certificate itself. Subsequent invocations of this function may assume that the certificate is already good because the policy is updated, which will not be the case for the impacted certificates due to this issue.

Reply:
This is not actually necessary - the `bundle` in the existing conditional comes from the cert itself, not the policy.